lib/pat.c

```diff
@@ -159,6 +159,15 @@ pat_node_get_left(grn_pat *pat, pat_node_common *node)
 #define PAT_DEL(x) ((x)->bits & PAT_DELETING)
 #define PAT_IMD(x) ((x)->bits & PAT_IMMEDIATE)
 #define PAT_LEN(x) (uint32_t)(((x)->bits >> 3) + 1)
+static inline uint32_t
+pat_node_get_key_length(grn_pat *pat, pat_node_common *node)
+{
+  if (pat_is_key_large(pat)) {
+    return (uint32_t)((node->node_large.bits >> 3) + 1);
+  } else {
+    return (uint32_t)((node->node.bits >> 3) + 1);
+  }
```

Comment:
`(bits >> 3) + 1)` という `bits` に対する処理が共通しているので、

```suggestion
  uint16_t bits;
  if (pat_is_key_large(pat)) {
    bits = node->node_large.bits;
  } else {
    bits = node->node.bits;
  }
  return (uint32_t)((bits >> 3) + 1);
```

みたいに書くと同じ処理をしているとわかりやすくなるのでお好みなんですが、どちらが良いのだろうか。
これだとbitsへの代入が増えるから良くないのかしら。